lib/sidekiq/redeploy/loader.rb

```diff
@@ -126,32 +129,47 @@ def handle_signal(signal)
         when USR2
           @reload_sidekiq = true
         when TTIN
-          ::Process.kill(signal, @sidekiq_pid)
+          @sidekiq_pids.each do |pid|
+            ::Process.kill(signal, pid)
+          end
         when TERM, INT
           @exit_loader = true
         end
       end
 
       def debug_handler(signal)
-        log_data = { signal:, current_pid: ::Process.pid, loader_pid: @loader_pid, sidekiq_pid: @sidekiq_pid,
+        log_data = { signal:, current_pid: ::Process.pid, loader_pid: @loader_pid, sidekiq_pids: @sidekiq_pids,
                      reload_sidekiq: @reload_sidekiq, exit_loader: @exit_loader }
         puts "handle_signal called with #{log_data}"
       end
 
-      def stop_sidekiq(pid)
-        ::Process.kill(TERM, pid) if pid
+      def stop_sidekiq(pids)
+        pids.each do |pid|
+          ::Process.kill(TERM, pid)
+        end
+        @sidekiq_pids = []
       end
 
-      def quiet_sidekiq(pid)
-        ::Process.kill(TSTP, pid) if pid
+      def quiet_sidekiq(pids)
+        pids.each do |pid|
+          ::Process.kill(TSTP, pid)
+        end
       end
 
-      def process_died?(pid)
+      def process_died?(pids)
```

Comment:
I wonder if it would be cleaner if we rename this to something like `running_pids` and have it return the running pids.

Something like this:
```
def running_pids(pids)
        return false if @exit_loader

        pids.each_with_object([]) do |pid, new_pids|
          if ::Process.waitpid(pid, ::Process::WNOHANG)
            num_dead += 1
          else
            new_pids << pid
          end
        end
      end
```